### DISCUSSION TITLE
Pi 5 & Bookworm

### DISCUSSION THREAD

**Discussion by Marliwahoo** [VirusPilot/stratux-pi4#20] [General]
I understand there are challenges with bookworm's networking and gpio management. Is there any information about an update to stratux using bookworm and/or the pi 5?

**Comment by VirusPilot** [comment]
I don't have a Pi5 for testing but would not expect any issues there. I have a Stratux running on bookworm, only minor adjustments are necessary (all boot related files have been moved in bookworm to /boot/firmware which conflicts with the expected location for some of our Stratux files).

**Reply by Marliwahoo** [reply]
I have it running on bookworm and pi 5. ADSB traffic, gps, AHRS and weather work fine.  Issues: Networking works only as default setting. It cannot be changed. I assume it has something to do with new networkmanager and no dhcpcd in Bookworm. BMP280 not working.

**Comment by brownleeb** [comment]
I loaded the stratux-pi4 version on a bookworm Pi4 yesterday and am still working through testing.
"AP+Client" wifi mode worked fine at first, then I did something that broke it.  During debugging, I found that the stratux wasnt making dhcp requests to my server (pi running dnsmasq).  I was able to get in by connecting to the stratux wifi AP.  I tried manually setting an IP address on wlan0, and I was able to ping, but couldnt log in.  On a whim, I changed the setting using the webUI back to AccessPoint, "submit wifi changes" and waiting a bit for the reconnect, then selecting AP+Client and saving again.  I think this might properly restart the DHCP process using the old. non-NetworkManager method and seems to work.
Everything seems to be working fine, GPS, wifi, ADSB.  I havent been out to check TIS-B

**Comment by VirusPilot** [comment]
Alight, good to know. Is your BMP280 working now?

**Reply by brownleeb** [reply]
No AHRS.  Just never got around to adding it....
I generally only use the stratux for traffic and TIS-B (on Avare) and any "time to dest" etc, I get off the mounted Garmin 430W

**Comment by Marliwahoo** [comment]
I got my bmp280 working by soldering the addr connector on the back of the bmp280 board. I'm using the Adafruit board with I2c connectors. It worked fine before the upgrade to bookworm. But with Bookworm it wasn't being recognized.
